### PR TITLE
Add tasklist component

### PR DESCRIPTION
**Pull request by thomasiles**
# Tasklist component

This branch cleans up the current view_component configuration and implements a single new component, the Tasklist.

It's based off the one in the DfE Register a trainee teach project but with a slightly modified style to suite the GDS pattern.

The Tasklist is not integrated into the application itself, you can see it under the component preview page at /preview

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [x] I've updated the documentation in (If any documentation requires updating)
    - [ ] README.md
    - [ ] Elsewhere (please link)

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -100,3 +100,7 @@ en:
     body: Contact the GOV.UK Forms team if you need to make changes to your forms or speak to someone about the service.
     title: Sorry, the service is unavailable
   skip_to_main_content: Skip to main content
+  task_statuses:
+    completed: completed
+    connot_start: cannot start yet
```
Comment (DavidBiddle) [inline config/locales/en.yml:105]: Typo:
```suggestion
    cannot_start: cannot start yet
```
(also appears in a few other files)

**Review by DavidBiddle — COMMENTED** [review]
Think this looks really good, have raised a comment about a typo but once that's resolved I'm happy for this to be merged! :tada

**Review by DavidBiddle — APPROVED** [review]

File: Gemfile.lock [hunk 2]
```diff
@@ -110,24 +110,24 @@ GEM
...
-    faraday (2.3.0)
-      faraday-net_http (~> 2.0)
+    erubi (1.11.0)
+    faraday (2.5.2)
+      faraday-net_http (>= 2.0, < 3.1)
       ruby2_keywords (>= 0.0.4)
-    faraday-net_http (2.0.3)
-    gds-sso (16.1.0)
+    faraday-net_http (3.0.0)
+    gds-sso (17.0.0)
```
Comment (aliuk2012) [inline Gemfile.lock:128]: We seem to be bumping more than just govuk_components/ view_components gem

File: config/application.rb [hunk 3]
```diff
@@ -19,5 +19,10 @@ class Application < Rails::Application
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
     config.exceptions_app = routes
+
+    config.view_component.preview_paths = [Rails.root.join("spec/components")]
+    config.view_component.preview_route = "/preview"
+    # Replace with value which will be true in local dev and PAAS dev
+    config.view_component.show_previews = !Rails.env.production?
```
[the remark this answers is not in the document]
Comment (thomasiles) [reply config/application.rb:26]: Yes - I totally forgot about this, I'll check it out in the morning.

File: Gemfile.lock (hunk 2 above)
Comment (thomasiles) [reply Gemfile.lock:128]: Hmm - I did have issues which meant I deleted the lock file and installed again. I'll go through them in detail.

Good catch!

Comment (aliuk2012) [reply Gemfile.lock:128]: could potentially, do it as a separate pr but its up to you.

File: config/application.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavidBiddle) [reply config/application.rb:26]: Think you want something to the effect of `PAAS_ENVIRONMENT.blank? || PAAS_ENVIRONMENT == 'dev'`

**Review by aliuk2012 — APPROVED** [review]
Great work @thomasiles, lets get this shipped

**Pull request merged**